### DISCUSSION TITLE
Can use MariaDB?

### DISCUSSION THREAD

**Discussion by Raphael909** [exadel-inc/CompreFace#546] [Q&A] [answered]
Hello Devs, 
First of all, thank you for this fantastic software.
I have couple of questions
1. can MariaDB replace postgresql .
2. can I still replace/rename subject using PUT if it was created directly while training faces. (I tried it but was seeing error in logs . HTTP status 404)

**Comment by pospielov** [comment, marked as answer]
1. I think all you need is to change is a configuration of these files:
https://github.com/exadel-inc/CompreFace/blob/master/java/api/src/main/resources/application.yml
https://github.com/exadel-inc/CompreFace/blob/master/java/admin/src/main/resources/application.yml
and of course, look how some of these variables are defined:
https://github.com/exadel-inc/CompreFace/blob/master/docs/Configuration.md
Then you need to build your own build, and I think it will work. But we haven't checked it. 
2. It's impossible to rename in 0.5.1 version, but we will release this functionality with the next release, we already added it to documentation:
https://github.com/exadel-inc/CompreFace/blob/master/docs/Rest-API-description.md#rename-a-subject
But as soon as you will need to build your build for MariaDB, I think you can do it from `master` branch, so your build will have renaming subject functionality as well
